language/tools/vm-genesis/src/lib.rs

```diff
@@ -351,18 +351,22 @@ fn create_and_initialize_owners_operators(
 /// Publish the standard library.
 fn publish_stdlib(session: &mut Session<StateViewCache>, stdlib: Modules) {
     let dep_graph = stdlib.compute_dependency_graph();
-    for module in dep_graph.compute_topological_order().unwrap() {
-        let module_id = module.self_id();
-        if module_id.name().as_str() == GENESIS_MODULE_NAME {
-            // Do not publish the Genesis module
-            continue;
-        }
-        let mut bytes = vec![];
-        module.serialize(&mut bytes).unwrap();
-        session
-            .publish_module(bytes, *module_id.address(), &mut GasStatus::new_unmetered())
-            .unwrap_or_else(|e| panic!("Failure publishing module {:?}, {:?}", module_id, e));
-    }
+    let mut addr = None;
+    let modules = dep_graph
+        .compute_topological_order()
+        .unwrap()
+        .map(|m| {
+            addr = Some(m.self_id().address().clone());
+            let mut bytes = vec![];
+            m.serialize(&mut bytes).unwrap();
+            bytes
+        })
+        .collect::<Vec<Vec<u8>>>();
+    // TODO: allow genesis modules published under different addresses. supporting this while
+    // maintaining the topological order is challenging.
+    session
+        .publish_module_bundle(modules, addr.unwrap(), &mut GasStatus::new_unmetered())
```

Comment:
@mengxu-fb: does this API need to take the sender address as input? It seems like it could read it off the first module and then complain unless all modules have the same address, but I might be missing something.

Reply:
I think the `sender` argument comes from the transaction sender. This is trying to prevent the case where user `0x2` tries to publish module with address `0x1`.

Reply:
Ah, got it--that makes sense. The call site here is an unusual one in that it doesn't go through the adapter and thus doesn't have a sender. Thanks for the clarification!